src/openshift/openshiftItem.ts

```diff
@@ -32,16 +32,42 @@ export abstract class OpenShiftItem {
         return new Promise((res) => setTimeout(res, timeout));
     }
 
+    static validateUniqueName(data: Array<OpenShiftObject>, value: string) {
+        const openshiftObject =  data.find((openshiftObject) =>  openshiftObject.getName() === value);
+        return openshiftObject && `This name is already used, please enter different name.`;
+    }
+
+    static async getName(message: string, data: Array<OpenShiftObject>) {
```

Comment:
please ensure the methods have defined return types

Reply:
done

Reply:
just string will do, undefined is a subtype of all other types anyway

Reply:
done